src/backend/joanie/core/models/products.py

```diff
@@ -887,7 +887,13 @@ def get_or_generate_certificate(self):
             not self.product.certificate_definition
             or self.product.type not in enums.PRODUCT_TYPE_CERTIFICATE_ALLOWED
         ):
-            return None, False
+            raise ValidationError(
```

Comment:
A validation error is not the exception that should be raise. The comment about this exception is `An error while validating data.`. IMO a dedicated exception should be created for the certificate generation.

Reply:
And you should use it like this IMO

```
            raise ValidationError({
                "product": _(
                    f"Product {self.product.title} does not allow to generate a certificate."
                )
            })
```

Reply:
According to our discussion, I have implemented a custom Exception class `CertificateGenerationError`.

```python
class CertificateGenerationError(Exception):
    """
    Exception raised when the certificate does not get generated because the ordert does not meet
    every condition.
    """
    def __init__(self, message: str, code: str):
        """
        Prepare exception attributes 'message', 'code'.
        """
        super().__init__(message)
        self.code = code
        self.message = message

```

Reply:
Finally, this will be the lastest version : 
```
class CertificateGenerationError(Exception):
    """
    Exception raised when the certificate does not get generated because the ordert does not meet
    every condition.
    """
```